### ISSUE TITLE
Loading existing models lead to wrong outputs when using BiBNRNN

### ISSUE THREAD

**Issue by ticoneva** [NervanaSystems/neon#359]
Insert the following code at the end of the imdb_lstm.py example:
```python
import numpy as np

output = model.get_outputs(train_set)

model.save_params("imdb_lstm.pkl")
model2 = Model("imdb_lstm.pkl")
output2 = model2.get_outputs(train_set)

print(np.allclose(output,output2))
print(output[:2])
print(output2[:2])
```

Then run 
```sh
python imdb_lstm.py -e 1 --rlayer_type 'bibnrnn'
```

Output:
```
False
[[  7.33556924e-03   9.92664397e-01]
 [  9.99926448e-01   7.35012145e-05]]
[[  3.42866336e-03   9.96571302e-01]
 [  9.99946117e-01   5.38577078e-05]]
```

The problem is specific to `'bibnrnn'` as none of the other recurrent layer options suffer from the same error. In particular, `'birnn'` does give the same outputs after reloading, which seems to indicate that the problem lies in the batch normalization process.

If I fetch the batch-norm mean and variance from the layer:
```python
gmean = model.layers.layers[1].gmean.get()
gmean2 = model2.layers.layers[1].gmean.get()
gvar = model.layers.layers[1].gvar.get()
gvar2 = model2.layers.layers[1].gvar.get()

print("gmeans:")
print(gmean[:2])
print(gmean2[:2])

print("gvars:")
print(gvar[:2])
print(gvar2[:2])
```

This is the output:
```
gmeans:
[[  3.50755217e-05]
 [ -1.53262044e-05]]
[[ 0.]
 [ 0.]]
gvars:
[[  6.59047291e-05]
 [  5.67276984e-05]]
[[ 0.]
 [ 0.]]
```

gmean and gvar are both zero when reloaded, resulting in different outputs.  

As far as I can tell, gmean and gvar are not saved when `save_params()` is called. I can get correct results if I add a `get_description()` inside `BiBNRNN` to save them:
```python
    def get_description(self, get_weights=False, keep_states=True):
        serial_dict = super(BiBNRNN, self).get_description(get_weights, keep_states)
        
        if get_weights:
            for key in ['gmean', 'gvar']:
                serial_dict['params'][key] = getattr(self, key).get()
        
        return serial_dict     
```
and make sure that they are not wiped in `init_params()`:
```python
    def init_params(self, shape):
...
        if self.gmean is None:
            self.gmean = self.be.zeros((nf, 1), dtype=self.stats_dtype,
                                      **self.get_param_attrs())
        if self.gvar is None:        
            self.gvar = self.be.zeros((nf, 1), dtype=self.stats_dtype,
                                      **self.get_param_attrs())
...
        self.inf_params = [self.gmean, self.gvar]
```

Now the results are identical before and after reload:
```
True
[[ 0.00225275  0.99774724]
 [ 0.99611974  0.00388023]]
[[ 0.00225275  0.99774724]
 [ 0.99611974  0.00388023]]
```
